polyply/src/graph_utils.py

```diff
@@ -183,3 +183,39 @@ def get_all_predecessors(graph, node, start_node=0):
             break
     predecessors.reverse()
     return predecessors
+
+def _find_longest_path(graph, source, target):
+    all_paths = list(nx.all_simple_edge_paths(graph, source, target=target))
+    max_path_length = 0
+    idx = 0
+    for jdx, path in enumerate(all_paths):
+        if len(path) > max_path_length:
+            max_path_length = len(path)
+            idx = jdx
+    return all_paths[idx]
```

Comment:
```suggestion
    all_paths = list(nx.all_simple_edge_paths(graph, source, target=target))
    return max(all_paths, key=len)
```
Unless you want to deal with multiple longest paths

Reply:
I want to deal with it in the sense that there can be more than one with equivalent path length. But the smaller code should account for that I guess?

Reply:
Yes. Then you should get the one with the lowest index, but don't pin me on that.